pkg/domain/infra/abi/secrets.go

```diff
@@ -17,15 +17,30 @@ func (ic *ContainerEngine) SecretCreate(ctx context.Context, name string, reader
 	if err != nil {
 		return nil, err
 	}
-	driverOptions := make(map[string]string)
 
+	// set defaults from config for the case they are not set by an upper layer
+	// (-> i.e. tests that talk directly to the api)
+	cfg, err := ic.Libpod.GetConfig()
```

Comment:
I still think defaults should be set only in `cmd/podman` and not here in the abi. Most commands have defaults set in `cmd/podman` and I'd like to keep that consistency and not read config files in the abi.

Reply:
Before this PR (in the old line 23) we checked if the driver was not specified and then set the hard coded default "file" since this was the only valid driver when this code was written, so we already set a default here. This is needed, because in `tests/apiv2/50-secrets.at` the very first test sends down `POST secrets/create Name=mysecret Data=c2VjcmV0` and expects that the secret is created. If we would not set defaults in the handler, this test would fail because the secret manager wouldn't know which driver to use. Since the new source for the default secret driver is no longer hard-coded but given in the (default) config, we need to reference it here. Otherwise we need to make the "driver" field in the payload required and this would be a very breaking change to the API.

Also I found 8 other uses of `ic.Libpod.GetConfig()` in the `abi` package.